functions/src/aqi-calculation/index.ts

```diff
@@ -161,8 +208,34 @@ exports.calculateAqi = functions.pubsub
       };
 
       currentData[purpleAirId] = currentSensorData;
+
+      // Update the AQI circular buffer for this element
+      const sensorDocRef = firestore.collection('/sensors').doc(docId);
+
+      const status =
+        knownSensor.data().aqiBufferStatus ?? bufferStatus.DoesNotExist;
+      // If the buffer status is In Progress we don't update the buffer at all
```

Comment:
Same comment as above

Reply:
It's not that we don't update the buffer at all, we just don't start updating the buffer again